compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -53,6 +53,34 @@ fn inline_attr<'ll>(cx: &CodegenCx<'ll, '_>, inline: InlineAttr) -> Option<&'ll
     }
 }
 
+#[inline]
+fn patchable_function_entry_attrs<'ll>(
+    cx: &CodegenCx<'ll, '_>,
+    attr: Option<PatchableFunctionEntry>,
+) -> SmallVec<[&'ll Attribute; 2]> {
```

Comment:
Since this is only ever two, instead of "two but sometimes it needs to be more", why not this?
```suggestion
) -> [Option<&'ll Attribute>; 2] {
```

At most it would add a `.filter_map(|x| x)` at the callsite. What do you think?

Reply:
I was thinking more in the "embed our expectations in the types" rather than as an optimization, but yeah, it does make things a bit messier I guess.